imap_processing/tests/ultra/unit/test_l2_utils.py

```diff
@@ -0,0 +1,102 @@
+"""Test creation of solid angle map."""
+
+import numpy as np
+import pytest
+
+from imap_processing.ultra.l2 import l2_utils
+
+# Parameterize with spacings (degrees here):
+valid_spacings = [0.1, 0.25, 0.5, 1, 5, 10, 20]
+invalid_spacings = [0, -1, 11]
+invalid_spacings_match_str = [
+    "Spacing must be positive valued, non-zero.",
+    "Spacing must be positive valued, non-zero.",
+    "Spacing must divide evenly into pi radians.",
+]
+
+
+@pytest.mark.parametrize("spacing", valid_spacings)
+def test_build_solid_angle_map(spacing):
+    """Test build_solid_angle_map function."""
+    solid_angle_map_steradians = l2_utils.build_solid_angle_map(
+        spacing, input_degrees=True, output_degrees=False
+    )
+    assert np.isclose(np.sum(solid_angle_map_steradians), 4 * np.pi, atol=0, rtol=1e-9)
+
+    solid_angle_map_sqdeg = l2_utils.build_solid_angle_map(
+        np.deg2rad(spacing), input_degrees=False, output_degrees=True
+    )
+    assert np.isclose(
+        np.sum(solid_angle_map_sqdeg), 4 * np.pi * (180 / np.pi) ** 2, atol=0, rtol=1e-9
+    )
+
+
+@pytest.mark.parametrize(
+    "spacing, match_str", zip(invalid_spacings, invalid_spacings_match_str)
+)
+def test_build_solid_angle_map_invalid_spacing(spacing, match_str):
+    """Test build_solid_angle_map function raises error for invalid spacing."""
+    with pytest.raises(ValueError, match=match_str):
+        _ = l2_utils.build_solid_angle_map(
+            spacing, input_degrees=True, output_degrees=False
+        )
+
+
+@pytest.mark.parametrize("spacing", valid_spacings)
+def test_build_az_el_grid(spacing):
+    """Test build_az_el_grid function."""
+    az_range, el_range, az_grid, el_grid = l2_utils.build_az_el_grid(
+        spacing=spacing,
+        input_degrees=True,
+        output_degrees=True,
+        centered_azimuth=False,
+        centered_elevation=True,
+    )
+
+    # Size checks
+    assert az_range.size == int(360 / spacing)
+    assert el_range.size == int(180 / spacing)
+    assert az_range.size == az_grid.shape[1]
+    assert el_range.size == el_grid.shape[0]
+
+    # Check grid values
+    expected_az_range = np.arange((spacing / 2), 360 + (spacing / 2), spacing)
+    expected_el_range = np.arange(-90 + (spacing / 2), 90 + (spacing / 2), spacing)[
+        ::-1
+    ]  # Note el order is reversed
+    assert np.allclose(az_range, expected_az_range), (
```

Comment:
`np.testing.assert_array_equal` (https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html) and `np.testing.assert_allclose` are preferred here as you don't need to add the f-strings. They will automatically print out discrepancies.